ChallengeBugTracking/Challenge3.cs

```diff
@@ -19,17 +19,19 @@ public class Leaf : Node {
     /// </summary>
     /// <param name="array">The array to flatten</param>
     public static string[] FlattenArray(IEnumerable<Node> nodes) {
-        // TODO: ISSUE #4242 : Our customer reported that this method doesn't work in the case of deeply nested arrays
-        // I added the test case to the test file, but I'm not sure how to fix it. Can you help?
-        var result = new List<string>();
+        List<string> output = new();
         foreach(var node in nodes) {
             if(node is Branch b) {
-                result.AddRange(b.Children.Cast<Leaf>().Select(l => l.Value));
+                foreach (var nodeChild in b.Children) {
+                    if (nodeChild is Leaf le)
+                        output.Add(le.Value);
+                        else
+                        output.AddRange(FlattenArray(b.Children.Where(r => r is Branch)));
```

Comment:
Something seems wrong here, and overall overly complex (why do you need a foreach in a foreach. There is a simpler way.

It is wrong here if I have `[["a", "b"], "c", ["d", "e"]]`, I am getting `["a", "b", "d", "e", "c", "a", "b", "d", "e"]`. Keep trying and you will find a way :).

<details>
  <summary>Hint</summary>
  
  Children is a collection to be flattened, just like the original collection. You don't need to iterate over it, just pass it recursively to your method.
</details>